Warn user about unused replace parameter in insert_rows (OracleHook)

RichieSK opened the pull request:
Following changes made to OracleHook class.

The replace boolean parameter claims to replace rows instead of insert, but this functionality has not been implemented and the documentation claims it works. This can be misleading to any user of this class. Furthermore, the parameter isn't even being used anywhere in this function.

In this change, I removed the unused 'replace' parameter in both the function and the docs to prevent any confusion amongst the users of this class.

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

RichieSK committed:
Remove unused replace boolean parameter in insert_rows function of OracleHook class

The replace boolean parameter claims to replace rows instead of insert, but this functionality has not been implemented and the documentation claims it works. This can be misleading to any user of this class. Furthermore, the parameter isn't even being used anywhere in this function.

In this commit, I removed the parameter to prevent any confusion amongst the users of this class.

Taragolis commented on airflow/providers/oracle/hooks/oracle.py:
```diff
@@ -251,8 +251,7 @@ def insert_rows(
         target_fields=None,
-        commit_every: int = 1000,
-        replace: bool | None = False,
+        commit_every: int = 1000
```
You can't remove `replace` argument, because it is comes from the `DbApiHook. insert_rows`

https://github.com/apache/airflow/blob/08f4b923ab6fe63aad72e3a9da9507ed5b9c6932/airflow/providers/common/sql/hooks/sql.py#L555-L565 

If you would like to inform user that it doesn't  take any affect for OracleHook you could inform by:
- Mentioned it into the docstring
- Warn users by appropriate warning message: `warning.warn("Foo bar do not use that", category=UserWarning, stacklevel=2)`

RichieSK replied:
I've reintroduced the replace parameter and implemented both suggestions to inform the user. Is the current version fine?

RichieSK committed:
Clarified usage of replace parameter in insert_rows

Re-introduced the replace boolean parameter.
Added a warning when the user sets the replace parameter to True.
Updated the docstring to let the user know that the replace boolean parameter does nothing.

RichieSK committed:
Updated formatting of warning in insert_rows

Updated the formatting of warning to increase readability.

romsharon98 reviewed (approved):
LGTM

dirrao reviewed (approved):
LGTM

syedahsn commented on airflow/providers/oracle/hooks/oracle.py:
```diff
@@ -272,8 +272,14 @@ def insert_rows(
             Default 1000, Set greater than 0.
             Set 1 to insert each row in each single transaction
-        :param replace: Whether to replace instead of insert
+        :param replace: Does not do anything.
```
I would leave the initial intended use-case of the parameter, and just mention that it is not currently implemented, like how you've done with the warning. Leaving it as `Does not do anything` will be confusing for anyone who sees it later on.

RichieSK replied:
Makes sense. I've updated it. Please lmk if the current version is fine.

RichieSK committed:
Update oracle.py

Switched back to original description of replace parameter with a warning of no current implementation

syedahsn reviewed (approved):
LGTM

RichieSK committed:
Format Fix

Added a format fix to pass static format testing
